miqa/core/rest/permissions.py

```diff
@@ -1,12 +1,58 @@
 from django.contrib.auth.models import User
 from django.db.models import Model
+from django.shortcuts import get_object_or_404
+from django.utils.functional import wraps
+from guardian.shortcuts import get_perms
 from rest_framework import status
 from rest_framework.exceptions import APIException
 from rest_framework.permissions import SAFE_METHODS, BasePermission
 from rest_framework.request import Request
+from rest_framework.response import Response
 from rest_framework.views import View
 
-from miqa.core.models import Experiment
+from miqa.core.models import Experiment, Project
+
+
+def project_permission_required(
+    read_access=True, review_access=False, superuser_access=False, **kwargs
```

Comment:
These arguments have 8 different permutations, but only three are used (or even meaningful). What do you think about having a different decorator for each permission level, like `@project_read_access_required` and `@project_review_access_required`?
Properly abstracting the different decorators would involve a function that takes a function and generates a decorator, which is perhaps just too silly, but it would make the API a little cleaner.

Incidentally, I feel like there is an existing decorator

Reply:
We can just take off `read_access` as an arg, since we assume it to be true and would never use this decorator with it as False. That brings our 8 permutations down to 4.